diesel_derives2/tests/helpers.rs

```diff
@@ -0,0 +1,72 @@
+use diesel::prelude::*;
+use std::sync::{Once, ONCE_INIT};
+
+cfg_if! {
+    if #[cfg(feature = "sqlite")] {
+        embed_migrations!("../migrations/sqlite");
+
+        pub type TestConnection = SqliteConnection;
+
+        fn connection_no_transaction() -> TestConnection {
+            SqliteConnection::establish(":memory:").unwrap()
+        }
+    } else if #[cfg(feature = "postgres")] {
+        extern crate dotenv;
+
+        embed_migrations!("../migrations/postgresql");
+
+        pub type TestConnection = PgConnection;
+
+        fn connection_no_transaction() -> TestConnection {
+            let database_url = dotenv::var("PG_DATABASE_URL")
+                .or_else(|_| dotenv::var("DATABASE_URL"))
+                .expect("DATABASE_URL must be set in order to run tests");
+            PgConnection::establish(&database_url).unwrap()
+        }
+    } else if #[cfg(feature = "mysql")] {
+        extern crate dotenv;
+
+        embed_migrations!("../migrations/mysql");
+
+        pub type TestConnection = MysqlConnection;
+
+        fn connection_no_transaction() -> TestConnection {
+            let database_url = dotenv::var("MYSQL_DATABASE_URL")
+                .or_else(|_| dotenv::var("DATABASE_URL"))
+                .expect("DATABASE_URL must be set in order to run tests");
+            MysqlConnection::establish(&database_url).unwrap()
+        }
+    } else {
+        compile_error!(
+            "At least one backend must be used to test this crate.\n \
+            Pass argument `--features \"<backend>\"` with one or more of the following backends, \
+            'mysql', 'postgres', or 'sqlite'. \n\n \
+            ex. cargo test --features \"mysql postgres sqlite\"\n"
+        );
+     }
+}
+
+static RUN_MIGRATIONS: Once = ONCE_INIT;
+
+pub fn connection() -> TestConnection {
+    let connection = connection_no_transaction();
+    RUN_MIGRATIONS.call_once(|| {
```

Comment:
This causes failures in all but the first test for sqlite, because each test is executed in a new database.
I would vote for simply calling `embedded_migrations::run()` unconditionally.

Reply:
I stuck an `if cfg!` in there. Running it unconditionally requires additional synchronization that would have been more painful.